cmake/ccache.cmake

```diff
@@ -17,15 +19,21 @@ if(NOT MSVC)
       )
       set(WITH_CCACHE ON)
     elseif(WITH_CCACHE)
-      list(APPEND CMAKE_C_COMPILER_LAUNCHER ${CCACHE_EXECUTABLE})
-      list(APPEND CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_EXECUTABLE})
+      foreach(lang IN ITEMS C CXX OBJCXX)
+        set(CMAKE_${lang}_COMPILER_LAUNCHER
+          ${CMAKE_COMMAND} -E env CCACHE_NOHASHDIR=1 ${CCACHE_EXECUTABLE}
```

Comment:
Will this [affect coverage](https://github.com/ccache/ccache/discussions/268) or is that an old issue that doesn't affect us anymore?